index.js

```diff
@@ -1,159 +1,102 @@
-		binaryCheck(el, null, cb);
-	});
-});
-
-checks.push(function home(cb) {
-	cb(null, {
-		title: process.platform === 'win32' ? '%USERPROFILE' : '$HOME',
-		message: !userHome && [
-			'environment variable is not set. This is required to know where',
-			'your home directory is. Follow this guide:',
-			'https://github.com/sindresorhus/guides/blob/master/set-environment-variables.md'
-		].join(' '),
-		fail: !userHome
-	});
-});
-
-checks.push(function node(cb) {
-	execFile(process.execPath, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
-
-		var version = stdout.trim();
-		var pass = semver.satisfies(version, '>=0.10.0');
-
-		cb(null, {
-			title: 'Node.js',
-			message: !pass && version + ' is outdated. Please update: http://nodejs.org',
-			fail: !pass
-		});
-	});
-});
-
-checks.push(function npm(cb) {
-	var bin;
-
-	try {
-		bin = which.sync('npm');
-	} catch (err) {
-		cb(null, {
-			title: 'npm',
-			message: 'Not installed. Please install Node.js (which bundles npm) from http://nodejs.org',
-			fail: true
+			throw err;
 		});
-		return;
-	}
+};
 
-	execFile(bin, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
+const binaryVersionCheck = bin => {
+	const title = bin;
+	const message = `Not installed. Please install it by running: npm install --global ${bin}`;
 
-		var localVersion = stdout.trim();
+	return whichP(bin)
+		.then(() => execa.stdout(bin, ['--version']))
+		.then(stdout => {
+			const localVersion = stdout.trim();
 
-		latestVersion('npm').then(function (version) {
-			var pass = semver.satisfies(localVersion, version);
+			return latestVersion(bin).then(version => {
+				const pass = semver.satisfies(localVersion, version);
 
-			cb(null, {
-				title: 'npm',
-				message: !pass && localVersion + ' is outdated. Please update by running: npm install --global npm',
-				fail: !pass
+				return {
+					title,
+					message: !pass && `${localVersion} is outdated. Please update by running: npm install --global ${bin}`,
+					fail: !pass
+				};
 			});
-		});
-	});
-});
-
-checks.push(function yo(cb) {
-	var bin;
+		})
+		.catch(err => {
+			if (/not found/.test(err.message)) {
+				return {
+					title,
+					message,
+					fail: true
+				};
+			}
 
-	try {
-		bin = which.sync('yo');
-	} catch (err) {
-		cb(null, {
-			title: 'yo',
-			message: 'Not installed. Please install it by running: npm install --global yo',
-			fail: true
+			throw err;
 		});
-		return;
-	}
-
-	execFile(bin, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
+};
 
-		var localVersion = stdout.trim();
+const home = {
+	title: process.platform === 'win32' ? '%USERPROFILE' : '$HOME',
+	message: !userHome && redent(`
+		environment constiable is not set. This is required to know where
+		your home directory is. Follow this guide:
+		https://github.com/sindresorhus/guides/blob/master/set-environment-constiables.md
+	`),
```

Comment:
Did a wrong assumption here, the text don't need to be splitted in 3 lines, it used `[].join(' ')` and not `[].join('\n')`. Could we solve that in a more clean way in ES6 instead of the array join method?